DDD-1.md

```diff
@@ -0,0 +1,179 @@
+```
+
+In order to achieve partition-awareness, we need to introduce an API and a set of components that do the following:
+
+1. Describe a *set* of source partitions to be processed by the task.
+2. Initialize the set above. Normally, based on the task configuration.
+3. Represent a source partition in higher-level terms specific to a given connector (e.g. database name).
+4. Allow mapping of connector-level partition (e.g. a database name) to the corresponding Kafka Connect source partition.
+5. Allow mapping of connector-level partition to the corresponding `OffsetContext`.
+6. Instead of a single `OffsetContext` per task, use a map of connector-level partitions to their offset context.
+
+### Support for multiple tasks per connector ([#9](https://github.com/sugarcrm/debezium/issues/9))
+
+Apart from being able to process multiple databases by a given task, it would be great if a connector was able to start multiple tasks and divide the databases between them. This way, it would allow to scale a single connector just by changing its `max.tasks` configuration.
+
+Being able to specify which subset of the databases a given task should process warrants the introduction of a new task-level configuration parameter ([#17](https://github.com/sugarcrm/debezium/issues/17)):
+
+| Name                  | Type     | Default | Importance | Doc                                                          |
+| :-------------------- | :------- | :------ | :--------- | :----------------------------------------------------------- |
+| `task.database.names` | `STRING` | `None`  | `HIGH`     | The list of comma-separated database names form which the given connector task should capture the changes from. Must be non-empty. |
+
+The connector can monitor the list of CDC-enabled databases and maintain the list of databases to capture the changes from ([#11](https://github.com/sugarcrm/debezium/issues/11)). This will require another configuration parameter:
+
+| Name            | Type      | Default | Importance | Doc                                                          |
+| :-------------- | :-------- | :------ | :--------- | :----------------------------------------------------------- |
+| `to.be.defined` | `BOOLEAN` | `False` | `Medium`   | If set to true, the connector will monitor the list of CDC-enabled databases and capture changes from all of them. |
+
+### Recover database schema from multiple partitions in one pass ([#14](https://github.com/sugarcrm/debezium/issues/14))
+
+Currently the internal schema representation used by the connector is capable of representing multiple databases (e.g. for the MySQL connector) but isn't capable of initializing from the database history of multiple partitions. It should be possible to initialize it multiple times, once for each partition but it may be suboptimal, especially if the database history topic and the number of partitions are large.
```

Comment:
But is it capable of mutlithreaded access?

Reply:
I assume, this question applies regardless of whether we want to initialize the schema for all partitions at once or sequentially. Even if we initialize the schema sequentially for each partition, multiple tasks running on the same worker may still access it in parallel. We'll have to check that additionally.

Reply:
As discussed offline, the SQL Server task will process each partition sequentially, so there is no need for the schema to be capable of handling multithreaded access.

I'm not familiar with how schema handling works in the MongoDB connector, so we may need to revisit this question if/when this approach is applied to the MongoDB connector.